File: Selenium/Project/Activity8.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;


public class Activity8 {
    WebDriver driver;

    @BeforeClass
    public void beforeMethod() {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        // Create the Actions object
        Actions builder = new Actions(driver);

        // Open the page
        driver.get("https://alchemy.hguy.co/jobs/");
        // Print the title of the page

    }
    @Test (priority = 8)
    public void Activity8() throws InterruptedException {
        driver.findElement(By.linkText("Post a Job")).click();
        driver.findElement(By.linkText("Sign in")).click();

        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        //wp-submit
        driver.findElement(By.id("wp-submit")).click();
    }
}

File: Selenium/Project/Activity8.py
import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By


# Create a fixture to initialize our webdriver
@pytest.fixture(scope="session")
def setup(request):
    # Initialize the webdriver
    driver = webdriver.Firefox()

    # Get the underlying collection
    session = request.node

    # Pass through the driver to the test class
    for item in session.items:
        cls = item.getparent(pytest.Class)
        setattr(cls.obj, "driver", driver)

    # Execute the test method
    yield

    # Close the browser once the test has ended
    request.addfinalizer(driver.close)


@pytest.mark.usefixtures("setup")
class TestSite:

    def test_navigation(self):
        self.driver.get("https://alchemy.hguy.co/jobs/wp-admin")

        username = self.driver.find_element(By.ID, "user_login")
        password = self.driver.find_element(By.ID, "user_pass")

        username.send_keys("root")
        password.send_keys("pa$$w0rd")

        login = self.driver.find_element(By.ID, "wp-submit")
        login.click()
        JobTitle = self.driver.title
        Expected_title = "Dashboard ‹ Alchemy Jobs — WordPress"

        assert JobTitle == Expected_title, "Title mismatch"


